### PR TITLE
[Caffe2] Remove MIOpen Softmax operator

### PR DESCRIPTION
**Pull request by ashishfarmer**
This PR contains changes for:
1. Removing MIOpen softmax operator. Will be added later with the required functionality
2. Enabling softmax_ops_test on ROCm target

cc: @bddppq

### CODE REVIEW

**Review by bddppq — APPROVED** [review]

**Comment by bddppq** [comment]
triggered caffe2 tests https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/1851/

**Comment by bddppq** [comment]
@ashishfarmer looks like softmax tests are flaky when running with rocm https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-test/347/console
